R/cohensFleissKappa.R

```diff
@@ -192,3 +193,59 @@ cohensFleissKappa <- function(jaspResults, dataset, options) {
   }
   return(jaspTable)
 }
+
+.computeKrippendorffsAlphaTable <- function(dataset, options, ready) {
+  # Create the JASP Table
+  jaspTable <- createJaspTable(title = gettextf("Krippendorff's alpha"))
+  jaspTable$addColumnInfo(name = "method", title = gettext("Method"), type = "string")
+  jaspTable$addColumnInfo(name = "kAlpha", title = gettext("Krippendorff's alpha"), type = "number")
+  jaspTable$position <- 2
+  
+  #dependencies
+  jaspTable$dependOn(
+    options = c(
+      "variables",
+      "krippendorffsAlpha",
+      "kappaIntervalOn",
+      "kappaConfidenceIntervalValue"
+    )
+  )
+  
+  formattedCIPercent <- format(
+    100 * options[["kappaConfidenceIntervalValue"]],
+    digits = 3,
+    drop0trailing = TRUE
+  )
+  
+  if (ready) {
+    #calculate Krippendorff's alpha
+    kAlphaData <- t(as.matrix(dataset))
+    method <- options[["alphaMethod"]]
+    kAlpha <- irr::kripp.alpha(kAlphaData, method)
+    
+    tableData <- list("method" = paste0(toupper(substr(method, 1, 1)), substr(method, 2, nchar(method))),
+                      "kAlpha" = kAlpha$value)
+    footnote <- gettextf('%i subjects/items and %i raters/measurements.', kAlpha$subjects, kAlpha$raters)
+    
+    if (options[["kappaIntervalOn"]]) {
+      bootstrapAlphas <- boot::boot(dataset, alpha.boot, R = 1000, method = method)
```

Comment:
How large are the data files for this analysis usually? I wonder if it makes sense to do the bootstrapping in another function, and save the samples the in the state, so when the confidence value is changed, the boot sample does not have to be created from scratch again... But if the data are usually small, then the speed gain could be negligible.

Reply:
Hm, I wouldn't know and either way it seems better if the analysis also runs smoothly for bigger datasets, so that sounds like a good idea.